mllib-local/src/main/scala/org/apache/spark/ml/linalg/BLAS.scala

```diff
@@ -39,8 +39,11 @@ private[spark] object BLAS extends Serializable {
   // For level-3 routines, we use the native BLAS.
   private[spark] def nativeBLAS: NetlibBLAS = {
     if (_nativeBLAS == null) {
-      _nativeBLAS =
-        try { NetlibNativeBLAS.getInstance } catch { case _: Throwable => javaBLAS }
+      _nativeBLAS = System.getProperty("spark.ml.allowNativeBlas", "true") match {
```

Comment:
This has to be a sys property because of how early it has to be initialized?

Reply:
Ideally, I think we should propagate the conf via `SparkConf`, and change the method signature to

```patch
- def nativeBLAS: NetlibBLAS
+ def nativeBLAS(allowNative: Boolean): NetlibBLAS
```

but I found many places call `BLAS.nativeBLAS` where `SparkConf` is unavailable, so I proposed to use sys property.